packages/dashmate/src/core/waitForCoreSync.js

```diff
@@ -4,26 +4,37 @@ const wait = require('../util/wait');
  * Wait Core to be synced
  *
  * @typedef {waitForCoreSync}
- * @param {RpcClient} rpcClient
- * @param {function(progress: number)} [progressCallback]
+ * @param {CoreService} coreService
+ * @param {function(progress: {percent: number, blocks: number, headers: number})} [progressCallback]
  * @return {Promise<void>}
  */
-async function waitForCoreSync(rpcClient, progressCallback = () => {}) {
+async function waitForCoreSync(coreService, progressCallback = () => {}) {
   let isSynced = false;
   let isBlockchainSynced = false;
   let verificationProgress = 0.0;
 
   do {
-    ({
-      result: { IsSynced: isSynced, IsBlockchainSynced: isBlockchainSynced },
-    } = await rpcClient.mnsync('status'));
-    ({
-      result: { verificationprogress: verificationProgress },
-    } = await rpcClient.getBlockchainInfo());
+    const info = await coreService.dockerContainer.inspect()
 
-    if (!isSynced || !isBlockchainSynced) {
-      await wait(10000);
-      progressCallback(verificationProgress);
+    try {
+
+      ({
+        result: {IsSynced: isSynced, IsBlockchainSynced: isBlockchainSynced},
+      } = await coreService.getRpcClient().mnsync('status'));
+
+      ({
+        result: {verificationprogress: verificationProgress, headers: headers, blocks: blocks},
+      } = await coreService.getRpcClient().getBlockchainInfo());
+
+      if (!isSynced || !isBlockchainSynced) {
+        await wait(10000);
+        progressCallback({percent: verificationProgress, headers, blocks});
+      }
+    } catch (e) {
+      if (info.State.Health.Status !== 'starting') {
```

Comment:
Shouldn't we check that service is started in a previous listr task and only after that wait for core to sync?